spec/models/mixins/supports_feature_mixin_spec.rb

```diff
@@ -200,4 +205,121 @@ def initialize(options = {})
       expect(NukeablePost.new(:bribe => false).supports_nuke?).to be true
     end
   end
+
+  describe ".subclasses_supporting" do
+    it 'detect' do
+      define_subclass("ProviderA", Post, :fake => true)
+      define_subclass("ProviderB", Post, :publish => false, :delete => true, :fake => true)
+
+      expect(Post.subclasses_supporting(:publish).map(&:name)).to eq(%w[ProviderA::Post])
+      expect(Post.subclasses_supporting(:delete).map(&:name)).to eq(%w[ProviderB::Post])
+      expect(Post.subclasses_supporting(:fake).map(&:name)).to match_array(%w[ProviderA::Post ProviderB::Post])
+    ensure
+      cleanup_subclass(Post)
```

Comment:
Can this be done via a normal `stub_const` call?  If not should this be a general `after` block at the top?

Reply:
so stub_const is successfully defining and cleaning up.
But rails caches the constants in the descendant tree.

After is tricky because we don't know what classes were defined?
I guess I could keep track of the classes defined and then clean those defined in the after test. But this is starting to get very picky and fragile.

Reply:
https://github.com/rspec/rspec-mocks/issues/1240

So they are having issues with descendant parser as well. They were also suggesting using Class.new (which doesn't play as well with `ActiveRecord` since it doesn't have a name but works ok if we set a table name, which we have to do with stub constant anyway.